### DISCUSSION TITLE
"Verify Address" button for Ledger Nano

### DISCUSSION THREAD

**Discussion by shufps** [iotaledger/firefly#7442] [Features / Enhancements]
According to Ledger, there should be a button in the UI that, when pressed, shows the current receiving address on the Ledger Nano.

Most of this task probably is UI because the iota-sdk and ledger.rs libs and the Ledger Nano app support it.

Generating a single address with `GenerateEd25519Addresses` and the bip-path of the current receiving address and the `ledger_nano_prompt` set to `true` should do it :thinking: 

```
/// Generate Ed25519 addresses.
GenerateEd25519Addresses {
    /// Addresses generation options
    options: GetAddressesOptions,
},
    
pub struct GetAddressesOptions {
    /// Coin type
    pub coin_type: u32,
    /// Account index
    pub account_index: u32,
    /// Range
    pub range: Range<u32>,
    /// Bech32 human readable part
    pub bech32_hrp: Hrp,
    /// Options
    pub options: Option<GenerateAddressOptions>,
}

pub struct GenerateAddressOptions {
    pub internal: bool,
    /// Display the address on ledger devices.
    pub ledger_nano_prompt: bool,
}
```

**Comment by begonaalvarezd** [comment]
Converted to issue https://github.com/iotaledger/firefly/issues/7443
